pkg/network/ovn_kubernetes.go

```diff
@@ -110,16 +125,27 @@ func fillOVNKubernetesDefaults(conf, previous *operv1.NetworkSpec, hostMTU int)
 	}
 
 	sc := conf.DefaultNetwork.OVNKubernetesConfig
-	// MTU is currently the only field we pull from previous.
-	// If it's not supplied, we infer it from  the node on which we're running.
+	// MTU  and GenevePort are currently the only fields we pull from previous.
+	// If MTU is not supplied, we infer it from  the node on which we're running.
 	// However, this can never change, so we always prefer previous.
 	if sc.MTU == nil {
 		var mtu uint32 = uint32(hostMTU) - 100 // 100 byte geneve header
-		if previous != nil && previous.DefaultNetwork.OVNKubernetesConfig != nil {
+		if previous != nil && previous.DefaultNetwork.OVNKubernetesConfig != nil &&
+			previous.DefaultNetwork.Type == operv1.NetworkTypeOVNKubernetes &&
+			previous.DefaultNetwork.OVNKubernetesConfig.MTU != nil {
```

Comment:
don't need this; we will have already validated that `DefaultNetwork.Type` didn't change

Reply:
ok